app/assets/javascripts/short-form/ShortFormApplicationController.js.coffee

```diff
@@ -557,10 +557,18 @@ ShortFormApplicationController = (
   $scope.validateHouseholdEligibility = (match) ->
     $scope.clearEligibilityErrors()
     form = $scope.form.applicationForm
+
+    if ShortFormApplicationService.householdDoesNotMeetAtLeastOneSeniorRequirement()
+      age = { minAge: $scope.listing.Reserved_community_minimum_age }
+      $scope.eligibilityErrors = [$translate.instant('ERROR.SENIOR_ANYONE', age)]
+      $scope.handleErrorState()
+      return
+
     # skip the check if we're doing an incomeMatch and the applicant has vouchers
     if match == 'incomeMatch' && $scope.application.householdVouchersSubsidies == 'Yes'
       ShortFormNavigationService.goToSection('Preferences')
       return
+
     ShortFormApplicationService.checkHouseholdEligibility($scope.listing)
```

Comment:
also confused what this function does because it checks income and household eligibility?

Reply:
Yes basically it refers to the callback after the income page:

```
    'income':
      scopedCallbacks: [{
        func: 'validateHouseholdEligibility'
        param: 'incomeMatch'
      }]
```

When the applicant puts in the income, we ask Salesforce if the householdsize-to-income ratio is valid for the listing so we call `validateHouseholdEligibility` again. The param 'incomeMatch' here tells the function that it's coming from the income page... BUT if the user said "Yes" on having a voucher subsidy then we skip it the income check.